src/main/java/Serializer.java

```diff
@@ -0,0 +1,40 @@
+import java.util.regex.Pattern;
+
+public class Serializer {
+    public String SerializeList(List list){
+        int listSize = list.getSize();
+        if(listSize==0){
+            return "";
+        }
+
+        StringBuilder result = new StringBuilder();
+        for(int i=0; i < listSize; i++){
+            result.append(list.getByIndex(i));
+            result.append(",");
+        }
+        result.deleteCharAt(result.lastIndexOf(","));
+
+        return result.toString();
+    }
+
+    private List DeserializeAsList(String value, List list){
```

Comment:
List DeserializeAsList(String value, List list) - this is a bad pattern. you are modifying an object, that is passed by link. it is much better to return a new list. 
https://stackoverflow.com/questions/40480/is-java-pass-by-reference-or-pass-by-value